libmachine/provision/utils.go

```diff
@@ -172,10 +174,17 @@ func ConfigureAuth(p Provisioner) error {
 		return err
 	}
 
-	// TODO: Do not hardcode daemon port, ask the driver
-	if err := utils.WaitForDocker(ip, dockerPort); err != nil {
-		return err
-	}
+	waitForDocker(p, dockerPort)
 
 	return nil
 }
+
+func waitForDocker(p Provisioner, dockerPort int) {
+	utils.WaitForSpecific(func() bool {
+		// HACK: Check netstat's output to see if anyone's listening on the Docker API port.
+		if _, err := p.SSHCommand(fmt.Sprintf("netstat -a | grep %d", dockerPort)); err != nil {
```

Comment:
Do you think it'd be better to use `netstat -l` directly here to make sure we don't have false positives?

Also, how consistent is the output of `netstat`?  Can we narrow down that `grep` further to get rid of even more false positives?  (like a Docker port of `253` matching something listening on `1253` for an obvious example of a problem edge case)